src/nvim/screen.c

```diff
@@ -1882,7 +1886,9 @@ static int line_putchar(LineState *s, schar_T *dest, int maxcells, bool rl)
   } else {
     if (p_arshape && !p_tbidi && arabic_char(u8c)) {
       // Do Arabic shaping.
-      int pc, pc1, nc;
+      int pc;
+      int pc1;
+      int nc;
```

Comment:
afaik we decided against this kind of change in the uncrustify discussions.

Reply:
what about multiple initializations on one line?

```c
int pc = 0, pc1 = 1, nc = 3;
```

IMO these are extremely error prone and ought to be split up. If it's the same rule that governs both then I think we should not do this (but if it's possible to behave differently for initialized and uninitialized declarations then I have no objection).

Reply:
i'm ok either way. @neovim/core +1 / -1

Reply:
@gpanders, I don't think the check can make that distinction: https://clang.llvm.org/extra/clang-tidy/checks/readability-isolate-declaration.html

It's quite useful for pointers, so I guess it depends how often do you think it could happen. 

One possibility is to fix the initializer but keep the others, and eventually turn off the check and leave it for the reviewers to enforce the rule.

Reply:
I'm strongly in favor of simply disallowing multiple declarations on one line.